src/modules/ToDoList.js

```diff
@@ -0,0 +1,113 @@
+    this.taskList = document.getElementById('task-list');
+  }
+
+  removeTask(index) {
+    this.tasks = this.tasks.filter((task) => this.tasks.indexOf(task) !== index);
+    this.storeData();
+  }
+
+  removeCompleted() {
+    const completedTasks = getCompleted();
+    let counter = 0;
+    completedTasks.forEach((task) => {
+      this.removeTask(task - counter);
+      counter += 1;
+    });
+    this.displayList();
+  }
+
+  displayList() {
+    this.taskList.innerHTML = '';
+    this.tasks.forEach((task) => {
+      task.index = this.tasks.indexOf(task);
+      const taskRow = document.createElement('div');
+      const taskMarkup = (task) => `
+                <input type="checkbox" id="accept-${task.index}" ${task.completed}>
+                <div><p class="task-description">${task.description}</p></div>
+                <i id="options-task-${task.index}" class="fas fa-ellipsis-v options"></i>
+                `;
+
+      taskRow.innerHTML = taskMarkup(task);
+      taskRow.classList.add('task-row');
+      taskRow.setAttribute('id', `task-row-${task.index}`);
+
+      taskRow.querySelector('div').addEventListener('click', () => {
+        this.editTask(taskRow, task.description, task.index);
+      });
+
+      const box = taskRow.querySelector('input[type=checkbox]');
+      box.addEventListener('change', (e) => {
+        if (e.target.checked) {
+          task.completed = 'checked';
+        } else {
+          task.completed = 'unchecked';
+        }
+        this.storeData();
+      });
+
+      this.taskList.appendChild(taskRow);
+    });
+  }
+
+  storeData() {
+    localStorage.setItem('tasks', JSON.stringify(this.tasks));
+  }
+
+  addTask() {
+    const task = new Task(this.tasks.length);
+    if ((this.tasks.length === 0 && task.description !== '')
+            || (JSON.stringify(this.tasks[this.tasks.length - 1].description)
+                !== JSON.stringify(task.description)
+                && task.description !== '')) {
+      this.tasks.push(task);
+      this.storeData();
+      this.displayList();
+      document.getElementById('new-task-description').value = '';
+    }
+  }
+
+  editTask(row, oldDescription, index) {
+    row.querySelector('i').classList.remove('fas', 'fa-ellipsis-vfas', 'options');
+    row.querySelector('i').classList.add('far', 'fa-trash-alt', 'remove');
+
+    row.querySelector('i').addEventListener('click', () => {
+      this.removeTask(index);
+      this.displayList();
+    });
+
+    const oldRows = document.querySelectorAll('.yellow');
+    oldRows.forEach((notSelected) => {
+      notSelected.classList.remove('yellow');
+    });
+
+    row.classList.toggle('yellow');
+    row.querySelector('div').innerHTML = `
+        <form id="update-task-form-${index}">
+        <input type="text" class="update-task-description" id="update-task-description-${index}" value="${oldDescription}" autofocus>
+        </form>
+        `;
```

Comment:
#### Keep It Simple, Smart
Every time that you press a task it changes the description tag between being a `<p>` and a `<input>`, I would recommend to just sticking with it being a `<input>` all the time. That way you reduce code lines and save DOM resources.

Reply:
Great insight!
Thanks for pointing it out, now I noticed that a big problem I encountered was caused because I changed the `<p>` to a `<input>`, using an input all the time might solve that.